iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/gcp/ComputeInstanceSshKeysCheck.java

```diff
@@ -0,0 +1,200 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.terraform.checks.gcp;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.common.api.checks.SecondaryLocation;
+import org.sonar.iac.common.api.tree.TextTree;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.common.checks.Trilean;
+import org.sonar.iac.common.extension.visitors.TreeContext;
+import org.sonar.iac.common.extension.visitors.TreeVisitor;
+import org.sonar.iac.terraform.api.tree.AttributeAccessTree;
+import org.sonar.iac.terraform.api.tree.BlockTree;
+import org.sonar.iac.terraform.api.tree.FileTree;
+import org.sonar.iac.terraform.api.tree.LiteralExprTree;
+import org.sonar.iac.terraform.api.tree.ObjectElementTree;
+import org.sonar.iac.terraform.api.tree.ObjectTree;
+import org.sonar.iac.terraform.checks.AbstractNewResourceCheck;
+import org.sonar.iac.terraform.symbols.AttributeSymbol;
+import org.sonar.iac.terraform.symbols.ResourceSymbol;
+
+import static org.sonar.iac.common.checks.PropertyUtils.get;
+import static org.sonar.iac.common.checks.PropertyUtils.valueOrNull;
+import static org.sonar.iac.terraform.api.tree.TerraformTree.Kind.BOOLEAN_LITERAL;
+import static org.sonar.iac.terraform.checks.utils.TerraformUtils.attributeAccessToString;
+
+@Rule(key = "S6405")
+public class ComputeInstanceSshKeysCheck extends AbstractNewResourceCheck {
+
+  private static final String MESSAGE = "Make sure that enabling project-wide SSH keys is safe here.";
+  private static final String OMITTING_MESSAGE = "Omitting metadata.block-project-ssh-keys enables project-wide SSH keys. Make sure it is safe here.";
+
+  private final TemplatesCollector collector = new TemplatesCollector("google_compute_instance_template");
+
+  @Override
+  public void initialize(InitContext init) {
+    super.initialize(init);
+    init.register(FileTree.class, (ctx, tree) -> collector.scan(new TreeContext(), tree));
+  }
+
+  @Override
+  protected void registerResourceConsumer() {
+    register("google_compute_instance",
+      resource -> checkMetadata(resource, true));
+
+    register("google_compute_instance_from_template",
+      this::checkInstanceFromTemplate);
+  }
+
+  private void checkInstanceFromTemplate(ResourceSymbol resource) {
+    AttributeSymbol srcTemplate = resource.attribute("source_instance_template");
+    srcTemplate.reportIfAbsent("Missing source_instance_template reference.");
+    if (srcTemplate.isPresent()) {
+      Tree srcTemplateValue = srcTemplate.tree.value();
+      if (!(srcTemplateValue instanceof AttributeAccessTree)) {
+        srcTemplate.ctx.reportIssue(srcTemplateValue, "Template reference of unexpected type.");
```

Comment:
Why do you think only an AttributeAccess is a valid references type? And even if this would be the case, why we should raise semantic invalid code. It's a rule to detect a specific security hotspot.